uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Upgrader/Step/1018.php

```diff
@@ -26,17 +26,17 @@ public function upgrade_1018 () {
     $query = "SELECT leaveRequest.id
       FROM {$leaveRequestTable} leaveRequest
       INNER JOIN {$absenceTypeTable} absenceType ON leaveRequest.type_id = absenceType.id
-      WHERE
-        absenceType.calculation_unit = {$calculationUnits['hours']}
-          AND leaveRequest.from_date IS NOT NULL
-          AND leaveRequest.from_date = leaveRequest.to_date
-          AND leaveRequest.is_deleted = 0";
+      WHERE absenceType.calculation_unit = {$calculationUnits['hours']}
+        AND leaveRequest.from_date IS NOT NULL
+        AND leaveRequest.from_date = leaveRequest.to_date
+        AND leaveRequest.is_deleted = 0";
 
     $leaveRequestRecord = CRM_Core_DAO::executeQuery($query);
 
     while ($leaveRequestRecord->fetch()) {
       $leaveRequest = LeaveRequest::findById($leaveRequestRecord->id);
-      $workDay = ContactWorkPattern::getContactWorkDayForDate(
+      $contactWorkPatternService = new ContactWorkPatternService();
```

Comment:
Please instantiate the service outside the while loop so that we won't have to be creating an object each time the loop runs.